From 0ec0581b4871a3161dcca2b6f6aa1d8296580402 Mon Sep 17 00:00:00 2001
From: Igor Grinberg <igrinberg@habana.ai>
Date: Sun, 05 May 2019 17:25:44 +0300
Subject: [PATCH] [SW-0] pll: export/change the hl_pll_bypass_get()

The hl_pll_bypass_get() is used internally by the PLL code.
As a preparation for adding the Gaudi W/A of the IF/MESH PLL init
issue, export the function for other modules and change the first
parameter to just PLL base offset value.

Change-Id: Idf423963fab7a1519def8ff6a256ea5c09904bbf
Signed-off-by: Igor Grinberg <igrinberg@habana.ai>
---

diff --git a/common/pll.c b/common/pll.c
index 961ff01..c55082b 100644
--- a/common/pll.c
+++ b/common/pll.c
@@ -9,12 +9,11 @@
 #include "uart.h"
 #include "pll.h"
 
-static enum hl_pll_bypass_val hl_pll_bypass_get(struct hl_pll_cfg *cfg,
-							enum hl_pll_div_out out)
+enum hl_pll_bypass_val hl_pll_bypass_get(u64 pll_base, enum hl_pll_div_out out)
 {
 	int offset = out * sizeof(u32);
 
-	return regmap_readl(cfg->base + HL_PLL_DIV_SEL + offset);
+	return regmap_readl(pll_base + HL_PLL_DIV_SEL + offset);
 }
 
 static void hl_pll_udelay(unsigned int usec)
@@ -31,7 +30,7 @@
 		 * 1) the CPU PLL offset in the array is 0
 		 * 2) DIV1 is the PLL output to timestamp counter.
 		 */
-		bval = hl_pll_bypass_get(hl_pll_cfg, HL_PLL_DIV1);
+		bval = hl_pll_bypass_get(hl_pll_cfg->base, HL_PLL_DIV1);
 	}
 
 	if (bval == HL_PLL_REF_CLK || bval == HL_PLL_DIV_REF)
@@ -215,7 +214,7 @@
 
 	for (i = 0; i < pll_num; i++) {
 		/* DIV0 is the undivided PLL output */
-		bval = hl_pll_bypass_get(hl_pll_cfg + i, HL_PLL_DIV0);
+		bval = hl_pll_bypass_get(hl_pll_cfg[i].base, HL_PLL_DIV0);
 		if (hl_pll_cfg[i].base == hl_pll_get_cpu_pll_base_off() &&
 		    bval != HL_PLL_REF_CLK &&
 		    bval != HL_PLL_DIV_REF)
diff --git a/common/pll.h b/common/pll.h
index ca260c4..edea940 100644
--- a/common/pll.h
+++ b/common/pll.h
@@ -55,6 +55,8 @@
 	u32	div_factor[HL_PLL_DIV_SIZE];
 };
 
+enum hl_pll_bypass_val hl_pll_bypass_get(u64 pll_base,
+					 enum hl_pll_div_out out);
 int hl_pll_init(unsigned int *pll_partial_init);
 int hl_pll_get_cfg(struct hl_pll_cfg **hl_pll_cfg);
 u64 hl_pll_get_cpu_cycles_per_usec(void);
